test/integration/test-submit-custom-header.js

```diff
@@ -31,17 +44,15 @@ var server = http.createServer(function(req, res) {
 
   form
     .on('field', function(name, value) {
-      fieldsPassed--;
-      var field = FIELDS[name];
-      assert.ok(field);
-      assert.strictEqual(value, ''+field);
+      var field = FIELDS.shift();
+      assert.strictEqual(name, field.name);
+      assert.strictEqual(value, field.value+'');
     })
     .on('file', function(name, file) {
-      fieldsPassed--;
-      var field = FIELDS[name];
-      assert.ok(field);
-      assert.strictEqual(file.name, path.basename(field.path));
-      assert.strictEqual(file.type, mime.lookup(file.name));
+      var field = FIELDS.shift();
```

Comment:
Very much agree. Was not aware of `request` using this.
Thanks for merging.

Unrelated - Is there a better medium to contact you for future PR's without cluttering the comments? I also have some questions.